### PR TITLE
Add cmdlets for Availability Group Listener resource of SqlVirtualMachine

### PR DESCRIPTION
**Pull request by ranishan**
## Description
Adding cmdlets for Availability Group Listener.
Cmdlet Design Review: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/458

## Checklist

- [X] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [X] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [X] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [X] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/SqlVirtualMachine/SqlVirtualMachine/help/Update-AzAvailabilityGroupListener.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.dll-Help.xml
+Module Name: Az.SqlVirtualMachine
+online version: https://docs.microsoft.com/en-us/powershell/module/az.sqlvirtualmachine/set-azsqlvmconfiggroup
```
Comment (VeryEarly) [inline src/SqlVirtualMachine/SqlVirtualMachine/help/Update-AzAvailabilityGroupListener.md]: this link should point to another url

File: src/SqlVirtualMachine/SqlVirtualMachine/help/Get-AzAvailabilityGroupListener.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.dll-Help.xml
+Module Name: Az.SqlVirtualMachine
+online version:
```
[the remark this answers is not in the document]
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/help/Get-AzAvailabilityGroupListener.md]: Added url now. Thank you.

File: src/SqlVirtualMachine/SqlVirtualMachine/help/New-AzAvailabilityGroupListener.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.dll-Help.xml
+Module Name: Az.SqlVirtualMachine
+online version: https://docs.microsoft.com/en-us/powershell/module/az.sqlvirtualmachine/get-azsqlvmgroup
```
[the remark this answers is not in the document]
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/help/New-AzAvailabilityGroupListener.md]: Changed url. Thank you.

File: src/SqlVirtualMachine/SqlVirtualMachine/help/Remove-AzAvailabilityGroupListener.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
+---
+external help file: Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.dll-Help.xml
+Module Name: Az.SqlVirtualMachine
+online version: https://docs.microsoft.com/en-us/powershell/module/az.sqlvirtualmachine/new-azsqlvmgroup
```
[the remark this answers is not in the document]
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/help/Remove-AzAvailabilityGroupListener.md]: Changed url. Thank you.

File: src/SqlVirtualMachine/SqlVirtualMachine/help/Update-AzAvailabilityGroupListener.md (hunk 1 above)
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/help/Update-AzAvailabilityGroupListener.md]: Changed url. Thank you.

**Comment by ranishan** [comment]
> please also add test cases for new classes

Added test cases. Since the setup requires manual steps, I have recorded the tests with manually set resources

**Pull request merged**